ksqldb-cli/src/main/java/io/confluent/ksql/cli/console/UnclosedQuoteChecker.java

```diff
@@ -0,0 +1,48 @@
+package io.confluent.ksql.cli.console;
+
+/**
+ * Checks to see if the line is in the middle of a quote.  Unlike
+ * org.jline.reader.impl.DefaultParser, this is comment aware.
+ */
+public class UnclosedQuoteChecker {
+  private static final String COMMENT = "--";
+
+  public static boolean isUnclosedQuote(String line) {
+    int quoteStart = -1;
+    for(int i = 0; line != null && i < line.length(); ++i) {
+      if (quoteStart < 0 && isQuoteChar(line, i)) {
+        quoteStart = i;
+      } else if (quoteStart >= 0 && line.charAt(quoteStart) == line.charAt(i) &&
```

Comment:
This `line.charAt(quoteStart) == line.charAt(i)` check if the char is a quote? Why not use `isQuoteChar(line, i)`?

Reply:
It's so you can have multiple quote types that contain the others e.g. "This could be 'valid'" and 'This could be "valid"'.

Given that we're just doing 'this type of quote', I'll simplify it.